product/views.py

```diff
@@ -1,59 +1,154 @@
-
-            return JsonResponse({"message" : "SUCCESS", "result" : filtered_products}, status = 200)
-            
+from .forms import ProductFullForm
+
+from user.models import OrderStatus, Address, User, Wishlist
+
+import boto3          
+import uuid   
+
+
+class ProductsView(View):
+    def get(self, request):
+        address_id = request.GET.get('address_id', None)
+
+        try:
+            if Product.objects.filter(address=address_id).exists():
+                products = Product.objects.filter(address=address_id).prefetch_related(Prefetch('productimage_set', queryset = ProductImage.objects.all(), to_attr='productimage_list'),
+                                                                                        Prefetch('wishlist_set', queryset = Wishlist.objects.all(), to_attr='wishlist_list'),
+                                                                                        Prefetch('productcomment_set', queryset = ProductComment.objects.all(), to_attr='comment_list'))   
+                                                                                        
+                result= [{
+                    "itemId"           : item.id,
+                    "imgSrc"           : item.productimage_list[0].image_url.url, 
+                    "title"            : item.name,
+                    "townName"         : (item.address.district+' '+item.address.neighborhood),
+                    "postedTime"       : item.updated_at.isoformat(),            
+                    "price"            : int(item.price),
+                    "wishCount"        : len(item.wishlist_list),
+                    "commentCount"     : len(item.comment_list),
+                    "viewed"           : item.viewed,
+                    "order_status"     : item.order_status.name,
+                }for item in products]
+
+                return JsonResponse({"message" : "SUCCESS", "productList" : result}, status = 200)
+            return JsonResponse({"message" : "NO_PRODUCT"}, status = 400)
+        except ValueError:
+            return JsonResponse({'message':'INVALID'}, status = 404)
+
+
+    def post(self, request):  
+        address_id = request.GET.get("address_id", None)
+        form = ProductFullForm(request.POST or None, request.FILES or None)
+        files = request.FILES.getlist('image')
+
+        if form.is_valid():
+            uploader_ins         = User.objects.get(id=2) #데코레이터 적용 전 임시
+            name                 = form.cleaned_data['name']
+            description          = form.cleaned_data['description']
+            price                = form.cleaned_data['price']
+            access_range         = form.cleaned_data['access_range']
+            order_status_ins     = OrderStatus.objects.get(name="판매중")
+            product_category_ins = form.cleaned_data["product_category"]
+            address_ins          = Address.objects.get(id=address_id)
+            product_ins          = Product.objects.create(
```

Comment:
코드만 보고도 instance인지 알 수 있습니다~
```suggestion
            address          = Address.objects.get(id=address_id)
            product          = Product.objects.create(
```

Reply:
반영완료!